include/up-cpp/transport/UTransport.h

```diff
@@ -0,0 +1,206 @@
+	using ListenHandle = typename CallbackConnection::Handle;
+
+	/// @brief Register listener to be called when UMessage is received
+	///        for the given URI.
+	///
+	/// @param uri Resolved UUri for where messages are expected to
+	///            arrived via the underlying transport technology.
+	/// @param listener Callback to be called when a message is received
+	///                 at the given URI. The UMessage will be provided
+	///                 as a parameter when the callback is called.
+	///
+	/// @returns * OKSTATUS and a connected ListenHandle if the listener
+	///            was registered successfully.
+	///          * FAILSTATUS with the appropriate failure and an
+	///            unconnected ListenHandle otherwise.
+	[[nodiscard]] std::tuple<v1::UStatus, ListenHandle> registerListener(
+	    const v1::UUri& uri, ListenCallback&& listener);
+
+	/// @brief Register listener to be called when UMessage is received
+	///        for the given URI, filtered by message source.
+	///
+	/// @param uri Resolved UUri for where messages are expected to
+	///            arrived via the underlying transport technology.
+	/// @param listener Callback to be called when a message is received
+	///                 at the given URI. The UMessage will be provided
+	///                 as a parameter when the callback is called.
+	/// @param source_filter Resolved UUri for where messages are expected to
+	///                      have been sent from.
+	///
+	/// @returns * OKSTATUS and a connected ListenHandle if the listener
+	///            was registered successfully.
+	///          * FAILSTATUS with the appropriate failure and an
+	///            unconnected ListenHandle otherwise.
+	[[nodiscard]] std::tuple<v1::UStatus, ListenHandle> registerListener(
+	    const v1::UUri& uri, ListenCallback&& listener,
+	    const v1::UUri& source_filter);
+
+	/// @brief Gets the default source Authority and Entity for all clients
+	///        using this transport instance.
+	///
+	/// @returns Const reference to the default source.
+	[[nodiscard]] const v1::UUri& getDefaultSource() const;
+
+	virtual ~UTransport() = default;
+
+protected:
+	/// @brief Represents the callable end of a callback connection.
+	///
+	/// This is a shared_ptr wrapping a callbacks::Connection. The
+	/// Connection will automatically disconnect from the matching
+	/// handle when destroyed. The Connection can be called directly
+	/// to invoke the callback.
+	using CallableConn = typename CallbackConnection::Callable;
+
+	/// @brief Register listener to be called when UMessage is received
+	///        for the given URI.
+	///
+	/// The transport library is required to implement this.
+	///
+	/// @remarks If this doesn't return OKSTATUS, the public wrapping
+	///          version of registerListener() will reset the connection
+	///          handle before returning it to the caller.
+	///
+	/// @param uri Resolved UUri for where messages are expected to
```

Comment:
No more resolved or not and we settled on source_filter and sink_filter for the parameters